### PR TITLE
api: implement delete by tag

### PR DESCRIPTION
**Pull request by shimish2**
- Added image deletion by tag feature.

### CODE REVIEW

File: pkg/storage/storage.go [hunk 1]
```diff
@@ -588,11 +588,14 @@ func (is *ImageStore) DeleteImageManifest(repo string, reference string) error {
...
 	}
 
+	isTag := false
+
 	// as per spec "reference" can only be a digest and not a tag
 	digest, err := godigest.Parse(reference)
 	if err != nil {
-		is.log.Error().Err(err).Msg("invalid reference")
-		return errors.ErrBadManifest
+		is.log.Debug().Str("invalid digest: ", reference).Msg("storage: assuming tag")
```
Comment (darshilgit) [inline pkg/storage/storage.go:596]: Is there a good way to see if it's a tag?

should we at least check if it matches a pattern? If not well defined than something as simple as having characters around `:`

File: pkg/storage/storage.go [hunk 2]
```diff
@@ -588,11 +588,14 @@ func (is *ImageStore) DeleteImageManifest(repo string, reference string) error {
 		return errors.ErrRepoNotFound
 	}
 
+	isTag := false
+
 	// as per spec "reference" can only be a digest and not a tag
```
Comment (darshilgit) [inline pkg/storage/storage.go:593]: remove this

File: pkg/storage/storage.go (hunk 1 above)
Comment (shimish2) [reply pkg/storage/storage.go:596]: we can enforce a check to look for specific set of tags following this rule https://github.com/opencontainers/image-spec/blob/main/annotations.md#rules but as of now we do not set any restriction on tag during upload so if we want to enforce this check we should do same for upload also.

File: pkg/storage/storage.go (hunk 2 above)
Comment (shimish2) [reply pkg/storage/storage.go:593]: Sure will remove.

File: README.md [hunk 3]
```diff
@@ -17,6 +17,7 @@ https://anuvu.github.io/zot/
 * Uses [OCI image layout](https://github.com/opencontainers/image-spec/blob/master/image-layout.md) for image storage
   * Can serve any OCI image layout as a registry 
 * Supports [helm charts](https://helm.sh/docs/topics/registries/)
+* Supports deletion by tag also.
```
Comment (rchincha) [inline README.md:20]: "Support image deletion by tag"

File: pkg/storage/storage.go [hunk 4]
```diff
@@ -620,7 +623,19 @@ func (is *ImageStore) DeleteImageManifest(repo string, reference string) error {
...
+		if isTag {
+			tag, ok := m.Annotations[ispec.AnnotationRefName]
+			if ok && tag == reference {
+				is.log.Debug().Str("deleting tag", tag).Msg("")
+
+				digest = m.Digest
+
+				found = true
+
+				continue
```
Comment (rchincha) [inline pkg/storage/storage.go:635]: We really cannot have the same tag repeat in an index.json? or can we?

Comment (shimish2) [reply pkg/storage/storage.go:635]: No this loop is appending those manifests which we want to keep so we can't break from here .

File: README.md [hunk 5]
```diff
@@ -17,6 +17,7 @@ https://anuvu.github.io/zot/
 * Uses [OCI image layout](https://github.com/opencontainers/image-spec/blob/master/image-layout.md) for image storage
   * Can serve any OCI image layout as a registry 
 * Supports [helm charts](https://helm.sh/docs/topics/registries/)
+* Supports image deletion by tag.
```
Comment (rchincha) [inline README.md:20]: remove the trailing period, none of the above statements have it

Comment (shimish2) [reply README.md:20]: Ok

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**